### PR TITLE
1421 update logs for Unicode Nul character

### LINKED ISSUE #1421: Transaction metadata gets dropped by Unicode NUL character
**OS**
Your OS: Gentoo Linux

**Versions**
The `db-sync` version (eg `cardano-db-sync --version`):
cardano-db-sync 13.1.1.1 - linux-x86_64 - ghc-8.10
git revision ec5b22b4364cca9a866b20de3564c27f2c751969
PostgreSQL version: 15.3

**Build/Install Method**
The method you use to build or install `cardano-db-sync`: nix build

**Run method**
The method you used to run `cardano-db-sync` (eg Nix/Docker/systemd/none): manually from CLI

**Additional context**
I traced this to a workaround made for #297 , now in https://github.com/input-output-hk/cardano-db-sync/blob/master/cardano-db-sync/src/Cardano/DbSync/Era/Util.hs line 50. Obviously the workaround has come as is to production.

TLDR; PostgreSQL can not record `\0` in stringy types. JSON and Unicode encodings may contain `\0`. JSON formatted value is dropped altogether.

**Problem Report**

While running cardano-db-sync to initially populate the database, there are intermittent Warning log level messages:
`[db-sync-node:Warning:75] [2023-05-31 11:02:29.12 UTC] prepareTxMetadata: dropped due to a Unicode NUL character.`

The warning message is misleading. Tx metadata is not dropped, but simply the `json` column of the row is recorded as `null`. Metadata payload is still stored in CBOR format in the `bytes` column.

While this behaviour may not be critical, it can be a surprise to database consumer.  It might be a good idea to document in the schema.md that the `json` column has this limitation and that the `bytes` column records the object payload in CBOR encoding, if not producing some actual fix to this.

### LINKED ISSUE #1623: Issue with logs formatting after `[db-sync-node:Warning]` message
**OS**
Your OS: `Linux Mint 21.2 Cinnamon`
Kernel: `5.15.0-92-generic`

**Versions**
```
[workstat:cardano.node.basicInfo.version:Notice:5] [2024-02-06 12:52:03.01 UTC] 8.7.3
[workstat:cardano.node.basicInfo.commit:Notice:5] [2024-02-06 12:52:03.01 UTC] a4a8119b59b1fbb9a69c79e1e6900e91292161e7

```

```
[db-sync-node:Info:6] [2024-02-06 15:42:55.56 UTC] Version number: 13.2.0.0
[db-sync-node:Info:6] [2024-02-06 15:42:55.56 UTC] Git hash: c5d6b8a1da25fbc7f283a34b07ed75dc747131c6
```

**Build/Install Method**
The method you use to build or install `cardano-db-sync`:
`nix`

**Additional context**
Release check for `13.2.0.0`

**Problem Report**

I noticed during sync check on `preprod` environment that shortly after starting `epoch 63` message the warning logs `[db-sync-node:Warning]`  are displaced and not starting as it was in the past - each on a separate new line.

Log snippet and screenshot attached below.

```
[db-sync-node:Info:64] [2024-02-06 16:11:42.27 UTC] Starting epoch 63
[db-sync-node:Info:64] [2024-02-06 16:11:42.27 UTC] Insert Babbage Block: epoch 63, slot 25574407, block 824961, hash 673fa32353ea5021cb0e3a694d4bfaee201ec9c83118f53f511936fd36fb40b0
[db-sync-node:Warning:64] [2024-02-06 16:11:42.62 UTC] prepareTxMetadata: dropped due to a Unicode NUL character. {"0bd0e5cb74bc80f5e3efee08ff3e655ecfa7d7923ce675c2fb9bd5cf":{"\u0000\u0006C[db-sync-node:Warning:64] [2024-02-06 16:11:42.62 UTC] prepareTxMetadata: dropped due to a Unicode NUL character. {"0bd0e5cb74bc80f5e3efee08ff3e655ecfa7d7923ce675c2fb9bd5cf":{"\u0000\r[db-sync-node:Info:64] [2024-02-06 16:11:42.66 UTC] Insert Babbage Block: epoch 63, slot 25575156, block 825000, hash 38a44fd5b45c2e8f47c27bf2d654f0e2a8a0b6c3116ca754a3b4182114218631
[db-sync-node:Info:72] [2024-02-06 16:11:44.73 UTC] Asynchronously wrote a ledger snapshot to ledger-state/preprod_13_2_0_0_bootstrap_tx_out/25574391-d06e9fe2f2-62.lstate in 2.476885368s.
[db-sync-node:Info:64] [2024-02-06 16:11:47.20 UTC] Inserted 3389 EpochStake for EpochNo 64
[db-sync-node:Info:64] [2024-02-06 16:11:52.20 UTC] Insert Babbage Block: epoch 63, slot 25688668, block 830000, hash 9c846a26c7dd86f9ca271be681fdea69825660c2bd092c704f7bb1a10cf5261b
[db-sync-node:Warning:64] [2024-02-06 16:11:53.82 UTC] prepareTxMetadata: dropped due to a Unicode NUL character. {"b":["[db-sync-node:Warning:64] [2024-02-06 16:11:53.93 UTC] prepareTxMetadata: dropped due to a Unicode NUL character. {"b":["[db-sync-node:Warning:64] [2024-02-06 16:11:53.95 UTC] prepareTxMetadata: dropped due to a Unicode NUL character. {"b":["[db-sync-node:Warning:64] [2024-02-06 16:11:53.95 UTC] prepareTxMetadata: dropped due to a Unicode NUL character. {"json_metadata":{"b":["[db-sync-node:Warning:64] [2024-02-06 16:11:54.08 UTC] prepareTxMetadata: dropped due to a Unicode NUL character. {"b":["[db-sync-node:Warning:64] [2024-02-06 16:11:54.12 UTC] prepareTxMetadata: dropped due to a Unicode NUL character. ["{\"filename\":\"images.jpe","g\",\"data\":[\"[db-sync-node:Info:64] [2024-02-06 16:12:01.59 UTC] Insert Babbage Block: epoch 63, slot 25814413, block 835000, hash 4a92e1d9ac91bfefb45a709366c9d1c1c84781665f0118bfce3bb03d95eb6341
[db-sync-node:Info:64] [2024-02-06 16:12:08.14 UTC] Insert Babbage Block: epoch 63, slot 25940045, block 840000, hash 062b95f1a91e3dafc3f8206ba788feb5ca617c1629c0a166ad2897f8b8e1c989
[db-sync-node:Info:64] [2024-02-06 16:12:08.14 UTC] Pool Offchain metadata fetch: 2 results, 15 fetch errors
[db-sync-node:Info:64] [2024-02-06 16:12:11.09 UTC] Deleted 0 tx_out
[db-sync-node:Info:64] [2024-02-06 16:12:11.33 UTC] Removing old epoch boundary file ledger-state/preprod_13_2_0_0_bootstrap_tx_out/22982389-c9fb79d3b5-56.lstate
[db-sync-node:Info:64] [2024-02-06 16:12:11.33 UTC] Removing 3 orphaned rewards
[db-sync-node:Info:64] [2024-02-06 16:12:11.34 UTC] Validate Epoch Rewards: total rewards that become spendable in epoch 64 are 2648
[db-sync-node:Info:64] [2024-02-06 16:12:11.34 UTC] Persistant SQL Statement Cache size is 432
[db-sync-node:Info:64] [2024-02-06 16:12:11.34 UTC] 
```

image

### PR DESCRIPTION
**Pull request by Cmdv**
# Description

this fixes #1421 

The logs were false in that not the whole row was drop, but instead if parsing failed with a Unicode Nul character the specific column it was intended for is recorded as `nul`.

This also effected a couple of other places so those were updated too.

```
[db-sync-node:Warning:75] [2024-01-31 11:02:29.12 UTC]  prepareTxMetadata: Column 'json' in table 'metadata' was recorded as null, due to a Unicode NUL character found when trying to parse the json.
```

This also fixes #1623 as outputting the full json made logs quite perplexing.

# Checklist

- [ ] Commit sequence broadly makes sense
- [ ] Commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] Any changes are noted in the [changelog](https://github.com/IntersectMBO/cardano-db-sync/blob/master/cardano-db-sync/CHANGELOG.md)
- [ ] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) on version 0.10.1.0 (which can be run with `scripts/fourmolize.sh`)
- [ ] Self-reviewed the diff

# Migrations

- [ ] The pr causes a [breaking change](https://github.com/IntersectMBO/cardano-db-sync/blob/master/doc/migrations.md) of type a,b or c
- [ ] If there is a breaking change, the pr includes a database migration and/or a fix process for old values, so that upgrade is possible
- [ ] Resyncing and running the migrations provided will result in the same database semantically

If there is a breaking change, especially a big one, please add a justification here. Please elaborate
more what the migration achieves, what it cannot achieve or why a migration is not possible.

### CODE REVIEW

**Comment by sgillespie** [comment]
> Column 'json' in table 'metadata' was recorded as null, due to a Unicode NUL character found when trying to parse the json.

This is definitely more clear, but I wonder if it's a little too verbose

**Comment by Cmdv** [comment]
> This is definitely more clear, but I wonder if it's a little too verbose

how about

`The 'json' column in 'metadata' became null due to a Unicode NUL character in the JSON parsing process.`

**Review by sgillespie — APPROVED** [review]
